[16.0][REF] l10n_br_base: use vat field for CNPJ and extract l10n_br_cpf_code

rvalyi opened the pull request:
O objetivo desse PR é de se aproximar do modelo de dados da Odoo que desde a versão 16.0 (e até a v18) usa o vat para armazenar o CNPJ.

Na v14 botamos o vat como related do cnpj_cpf, a ideia é agora de usar a coluna vat no banco de dados e conservar o campo cnpj_cpf como related para conservar uma certa compatibilidade e facilitar os cherry-picks com o codigo da v14 por enquanto.

Eu tb introduzi o campo l10n_br_cpf_code pro CPF de acordo com o modelo do Odoo a partir da v16: 
https://github.com/odoo/odoo/blob/16.0/addons/l10n_br/models/res_partner.py

Eu tb aproveitei para melhorar a mensagem de validação.

mileo committed:
[NEW] Módulo de relatórios contabeis

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[NEW] Some fixes and report styles

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

DiegoParadeda committed:
[ADD] MIS Reports for simples nacional

mileo committed:
[NEW] DRE auto expand accounts

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[NEW] OCA readme files

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] DRE expression CSLL e IRPJ

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[REF] BP with new account.account.types

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] l10n_br_mis_report dependencies

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX

mileo committed:
[FIX] Mis Report

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] DRE e BP images

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

mileo committed:
[FIX] wrong bp expression

Signed-off-by: Luis Felipe Mileo <mileo@kmee.com.br>

oca-travis committed:
[UPD] Update l10n_br_mis_report.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_br_mis_report 12.0.1.1.0

rvalyi committed:
l10n_br_mis_report: fixed images

OCA-git-bot committed:
[UPD] README.rst

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

mileo committed:
[FIX] DRE deduções da receita

OCA-git-bot committed:
l10n_br_mis_report 12.0.1.2.0

rvalyi committed:
DRE: bale->balp everywhere

OCA-git-bot committed:
l10n_br_mis_report 12.0.1.3.0

rvalyi committed:
[IMP] l10n_br_mis_report: black, isort, prettier

rvalyi committed:
[MIG] l10n_br_mis_report: Migration to 14.0

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 0.0% (0 of 52 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_mis_report
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_mis_report/pt_BR/

mileo committed:
[REF] deprecade data xml node l10n_br_mis_report

OCA-git-bot committed:
l10n_br_mis_report 14.0.1.0.1

rvalyi committed:
[REF] cleaner db wise mixin mappings

mbcosta committed:
[FIX] l10n_br_fiscal: proper partner_id

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice/Fiscal.

[FIX] l10n_br_fiscal: Document get Fiscal Partner

mbcosta committed:
[FIX] l10n_br_sale: proper partner_id

Invoice should be create with partner_invoice_id not with partner_id.

[FIX] l10n_br_sale: Document get Fiscal Partner

[FIX] l10n_br_sale: Related Fiscal Partner

[FIX] l10n_br_sale: partner_invoice_id fix

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice, in case of Sale should it use the field partner_invoice_id.

mbcosta committed:
[FIX] l10n_br_stock_account: partner_shipping_id

when creating an Invoice use the Partner mapping by the methods of module instead of Fiscal Dict and inform partner_shipping_id with Partner of Picking instead in the case of it's different of Partner to Invoice.

CristianoMafraJunior committed:
[IMP] l10n_br_nfe: DANFE margins config

CristianoMafraJunior committed:
[IMP] l10n_br_account_nfe: print danfe

rvalyi committed:
[REF] extract l10n_br_fiscal_edi <- l10n_br_fiscal

[REF] extract l10n_br_fiscal_edi from l10n_br_fiscal 2/2

[TMP] leave invalidate.number in l10n_br_fiscal

[REF] l10n_br_fiscal: mv doc wrkflw ->fisc edi

[REF] l10n_br_fiscal_edi: mv doc wrkflw ->fisc edi

[WIP] mv edoc import to fiscal_edi

[REF] add fiscal document hook

[FIX] l10n_br_fiscal_edi:  wrong status_description

[FIX] l10n_br_fiscal: edi migration

mileo committed:
[MIG] l10n_br_mis_report: Migration to 15.0

mileo committed:
[IMP] l10n_br_mis_report: apply pre-commit auto fixes

mileo committed:
[MIG] l10n_br_mis_report: Migration to 16.0

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account_nfe/

antoniospneto committed:
[REF] l10n_br_account_nfe: rename nfce contingency function

OCA-git-bot committed:
[BOT] post-merge updates

antoniospneto committed:
[IMP] l10n_br_account_nfe: add unit test for nfe with IPI

OCA-git-bot committed:
[BOT] post-merge updates

antoniospneto committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 80.0% (28 of 35 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account_nfe/pt_BR/

Commit:
[UPD] Update l10n_br_account_nfe.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_account_nfe/

rvalyi committed:
[REF] spec_driven_model: simplify register hook

rvalyi committed:
[REF] l10n_br_nfe: adapt to simpler spec_driven_model hook

rvalyi committed:
[REF] l10n_br_account_nfe: lint

rvalyi committed:
[MIG] l10n_br_account_nfe: Migration to 15.0

rvalyi committed:
[IMP] l10n_br_account_nfe: pre-commit stuff

rvalyi committed:
[MIG] l10n_br_account_nfe: Migration to 16.0

rvalyi committed:
[REF] l10n_br_account: adapt to l10n_br_fiscal_edi

rvalyi committed:
[REF] l10n_br_nfse_focus: adapt to l10n_br_fiscal_edi

mbcosta committed:
[FIX] l10n_br_purchase: proper partner_id

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice, in the Purchase case the code use address_get to create Invoice.

[FIX] l10n_br_purchase: Document get Fiscal Partner

mbcosta committed:
[IMP] l10n_br_purchase: demo data and tests

for the cases where the Partner of the object can be or not the Partner to Invoice.

mbcosta committed:
[IMP] l10n_br_sale: demo data and tests

for the case where the Partner of the object can be or not the Partner to Invoice.

mbcosta committed:
[FIX] l10n_br_stock_account: proper partner_id

When mapping the Line Fiscal Operation and Taxes the Partner of the object can be or not the Partner to Invoice, in the Picking case the code use address_get to create Invoice.

[FIX] l10n_br_stock_account: Get Fiscal Partner

mbcosta committed:
[IMP] l10n_br_stock_account: demo data and tests

for the case where the Partner of the object can be or not the Partner to Invoice.

rvalyi committed:
[REF] l10n_br_nfe: depend on l10n_br_fiscal_edi

rvalyi committed:
[REF] l10n_br_nfse: depend on l10n_br_fiscal_edi

rvalyi committed:
[REF] l10n_br_fiscal_closing: depend on l10n_br_fiscal_edi

corredato committed:
[NEW] CNPJ search in CRM Lead

mbcosta committed:
[FIX] l10n_br_stock_account: Get Default OP Fiscal

rvalyi committed:
[REF] l10n_br_fiscal: document_fiscal* rename

rvalyi committed:
[REF] l10n_br_account: fiscal_document* rename

rvalyi committed:
[REF] l10n_br_sale: fiscal_document rename

rvalyi committed:
[REF] l10n_br_account_nfe: adapt to simpler hook

antoniospneto committed:
[MIG] l10n_br_account_nfe: improve tests

DiegoParadeda committed:
[IMP] l10n_br_nfe: set nfe40_IEST

antoniospneto committed:
[REF] l10n_br_mis_report: pre-commit update

antoniospneto committed:
[MIG] l10n_br_mis_report: replace type to prefix

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/

OCA-git-bot committed:
Merge PR #3273 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_account_nfe.pot

Commit:
[UPD] Update l10n_br_nfe.pot

Commit:
[UPD] Update l10n_br_nfe_spec.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe_spec
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe_spec/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/

antoniospneto committed:
l10n_br_cnpj_search: merge webservice classes

resolve pre-commit:
l10n_br_cnpj_search/models/serpro_webservice.py:19:
[R8180(consider-merging-classes-inherited), SerproWebservice]
Consider merging classes inherited to "l10n_br_cnpj_search.webservice.abstract" from l10n_br_cnpj_search/models/receitaws_webservice.py:11:4.

antoniospneto committed:
l10n_br_fiscal: merge tax_definition classes

antoniospneto committed:
l10n_br_ie_search: merge webservice classes

antoniospneto committed:
l10n_br_fiscal: merge ncm classes

mileo committed:
[FIX][l10n_br_nfe] exported xml not have all fields computed

Signed-off-by: Luis Felipe Miléo <mileo@kmee.com.br>

mileo committed:
[FIX][l10n_br_nfe] certificate company env -> doc

Signed-off-by: Luis Felipe Miléo <mileo@kmee.com.br>

DiegoParadeda committed:
[IMP] l10n_br_fiscal: tax.definition states usability

OCA-git-bot committed:
Merge PR #3329 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_fiscal.pot

Commit:
[UPD] Update l10n_br_account.pot

Commit:
[UPD] Update l10n_br_fiscal_edi.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

OCA-git-bot committed:
Merge PR #3350 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

OCA-git-bot committed:
Merge PR #3356 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (65 of 65 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_dfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_dfe/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 95.4% (21 of 22 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_payment_brcobranca
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_payment_brcobranca/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (38 of 38 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_nfe/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 90.3% (1159 of 1283 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 15.2% (119 of 778 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 85.5% (278 of 325 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_sale
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_sale/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 87.0% (263 of 302 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/pt_BR/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (303 of 303 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/pt_BR/

DiegoParadeda committed:
[FIX] l10n_br_nfe: export nfe40_iest to XML and PDF

antoniospneto committed:
[MIG] l10n_br_mis_report: enhance README

DiegoParadeda committed:
[IMP] cancel invoice on fiscal document denial

OCA-git-bot committed:
Merge PR #3343 into 16.0

Signed-off-by marcelsavegnago

Commit:
[UPD] Update l10n_br_mis_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 52.6% (158 of 300 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_cnab_structure
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_cnab_structure/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 90.4% (1160 of 1283 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 18.5% (169 of 913 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (65 of 65 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_dfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_dfe/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (75 of 75 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_certificate
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_certificate/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (96 of 96 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_closing
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_closing/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 25.1% (124 of 493 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_payment_order/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (40 of 40 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_zip
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_zip/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (364 of 364 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_hr
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_hr/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 86.7% (282 of 325 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_sale
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_sale/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (303 of 303 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 87.7% (265 of 302 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 0.7% (8 of 1135 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe_spec
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe_spec/pt_BR/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 22.3% (174 of 778 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/pt_BR/

antoniospneto committed:
[REF] l10n_br_account: remove dead code

OCA-git-bot committed:
Merge PR #3372 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

corredato committed:
[IMP] l10n_br_fiscal: icms fcp and difal destination comments

OCA-git-bot committed:
Merge PR #3308 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (22 of 22 strings)

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_payment_brcobranca
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_payment_brcobranca/pt_BR/

WesleyOliveira98 committed:
[FIX] l10n_br_account: fix _compute_fiscal_document_ids

DiegoParadeda committed:
[IMP] l10n_br_account: document deny tests

OCA-git-bot committed:
Merge PR #3378 into 16.0

Signed-off-by marcelsavegnago

OCA-git-bot committed:
[BOT] post-merge updates

DiegoParadeda committed:
[FIX] l10n_br_fiscal_edi: rename refactored reports

rvalyi committed:
[REF] l10n_br_base: CNPJ->vat; +l10n_br_cpf_code

fix

rvalyi committed:
[FIX] l10n_br_fiscal: fix field defaults/labels

[FIX] l10n_br_fiscal: default currency_id fix

rvalyi commented:
pessoal pensei num primeiro passo melhor para ter uma compatibilidade com l10n_br_cpf_code tb sem perder a facilidade dos cherry-picks com a v14. Novo PR em breve.

EDIT: acabei finalizando esse PR que nao é muito invasivo

rvalyi committed:
[REF] l10n_br_nfe: CNPJ->vat; +l10n_br_cpf_code

rvalyi committed:
[FIX] l10n_br_fiscal_edi: fix warnings

rvalyi committed:
[REF] l10n_br_cnpj_search: CNPJ->vat

fix

rvalyi committed:
[REF] l10n_br_ie_search: CNPJ->vat

rvalyi commented:
teste de migração. Dados da v16 antes do PR:

```
odoo16=# select id, vat, cnpj_cpf, is_company, name from res_partner where cnpj_cpf is not null;
 id | vat |      cnpj_cpf      | is_company |                   name
----+-----+--------------------+------------+-------------------------------------------
  1 |     | 97.231.608/0001-69 | t          | Sua Empresa
 41 |     | 86.144.802/0001-90 | t          | Intel do Brasil
 42 |     | 62.228.384/0001-51 | t          | AMD do Brasil
 43 |     | 92.737.124/0001-72 | t          | LG do Brasil
 44 |     | 21.717.475/0001-73 | t          | Samsung Recife
 45 |     | 77.256.611/0001-20 | t          | Positivo Informatica
 46 |     | 73.623.891/0001-06 | t          | Dell Computadores do Brasil
 47 |     | 14.685.381/0001-02 | t          | Info Bahia
 48 |     | 38.633.837/0001-40 | t          | Informatica Manaus
 49 |     | 81.493.979/0001-89 | t          | Cliente 1 SP Contribuinte
 50 |     | 12.046.835/0001-61 | t          | Cliente 2 -SP - Simples Nacional
 51 |     | 96.660.336/0001-50 | f          | Cliente 2 - SP - Endereço Entrega
 52 |     | 46.081.676/0001-58 | t          | Cliente 3 - Z/F Manaus - Contribuinte
 53 |     | 84.148.732/0001-13 | t          | Cliente 4 - Z/F Manaus - Simples Nacional
 54 |     | 14.753.702/0001-50 | t          | Cliente 5 - Contribuinte - PE
 55 |     | 97.449.840/0001-78 | t          | Cliente 6 - Simples Nacional- PE
 56 |     | 43.266.887/0001-77 | t          | Cliente 7 RS - Contribuinte
 57 |     | 13.691.522/0001-29 | f          | Cliente 7 - RS - Endereço Cobrança
 58 |     | 32.406.080/0001-76 | t          | Cliente 8 RS - Simples Nacional
 59 |     | 11.034.414/0001-58 | t          | Akretion Sao Paulo
 61 |     | 11.034.414/0003-10 | t          | Akretion Aluminio - SP
 62 |     | 26.355.289/0001-55 | t          | Cliente 9 MG - PJ Não Contribuinte
 63 |     | 276.288.360-11     | f          | Cliente 10 MG - PF Não Contribuinte
 64 |     | 23.130.935/0001-98 | t          | KMEE
 66 |     | 765.865.078-12     | f          | Cliente 11 SP - PF Não Contribuinte
 67 |     | 81.583.054/0001-29 | t          | Empresa Lucro Presumido
 68 |     | 59.594.315/0001-57 | t          | TESTE - Simples Nacional
(27 rows)
```

depois da migração:

```
odoo16=# select id, vat, l10n_br_cpf_code, is_company, name from res_partner where cnpj_cpf is not null;
 id |        vat         | l10n_br_cpf_code | is_company |                   name
----+--------------------+------------------+------------+-------------------------------------------
  1 | 97.231.608/0001-69 |                  | t          | Sua Empresa
 41 | 86.144.802/0001-90 |                  | t          | Intel do Brasil
 42 | 62.228.384/0001-51 |                  | t          | AMD do Brasil
 43 | 92.737.124/0001-72 |                  | t          | LG do Brasil
 44 | 21.717.475/0001-73 |                  | t          | Samsung Recife
 45 | 77.256.611/0001-20 |                  | t          | Positivo Informatica
 46 | 73.623.891/0001-06 |                  | t          | Dell Computadores do Brasil
 47 | 14.685.381/0001-02 |                  | t          | Info Bahia
 48 | 38.633.837/0001-40 |                  | t          | Informatica Manaus
 49 | 81.493.979/0001-89 |                  | t          | Cliente 1 SP Contribuinte
 50 | 12.046.835/0001-61 |                  | t          | Cliente 2 -SP - Simples Nacional
 51 | 96.660.336/0001-50 |                  | f          | Cliente 2 - SP - Endereço Entrega
 52 | 46.081.676/0001-58 |                  | t          | Cliente 3 - Z/F Manaus - Contribuinte
 53 | 84.148.732/0001-13 |                  | t          | Cliente 4 - Z/F Manaus - Simples Nacional
 54 | 14.753.702/0001-50 |                  | t          | Cliente 5 - Contribuinte - PE
 55 | 97.449.840/0001-78 |                  | t          | Cliente 6 - Simples Nacional- PE
 56 | 43.266.887/0001-77 |                  | t          | Cliente 7 RS - Contribuinte
 57 | 13.691.522/0001-29 |                  | f          | Cliente 7 - RS - Endereço Cobrança
 58 | 32.406.080/0001-76 |                  | t          | Cliente 8 RS - Simples Nacional
 59 | 11.034.414/0001-58 |                  | t          | Akretion Sao Paulo
 61 | 11.034.414/0003-10 |                  | t          | Akretion Aluminio - SP
 62 | 26.355.289/0001-55 |                  | t          | Cliente 9 MG - PJ Não Contribuinte
 64 | 23.130.935/0001-98 |                  | t          | KMEE
 67 | 81.583.054/0001-29 |                  | t          | Empresa Lucro Presumido
 68 | 59.594.315/0001-57 |                  | t          | TESTE - Simples Nacional
 63 |                    | 276.288.360-11   | f          | Cliente 10 MG - PF Não Contribuinte
 66 |                    | 765.865.078-12   | f          | Cliente 11 SP - PF Não Contribuinte
```

OCA-git-bot committed:
Merge PR #3400 into 16.0

Signed-off-by yvaucher

Commit:
[UPD] Update l10n_br_fiscal_edi.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3399 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3393 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3395 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3405 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

kaynnan committed:
[IMP] l10n_br_fiscal_closing: add document_cte_ids

rvalyi committed:
[FIX] l10n_br_account: avoid duplicated labels

rvalyi committed:
[FIX] l10n_br_purchase: fix field name

rvalyi committed:
[REM] l10n_br_fiscal_closing: rm useless onchange

rvalyi committed:
[MIG] l10n_br_nfe: deprecated flush and cache meth

rvalyi committed:
[REF] spec_driven_model: rm dead code; + XML tests

Tiago370 committed:
[IMP] l10n_br_cnpj_search: change vcr mocks for unittest mocks.

OCA-git-bot committed:
Merge PR #3385 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

Commit:
[UPD] Update l10n_br_account.pot

Commit:
[UPD] Update l10n_br_purchase.pot

Commit:
[UPD] Update l10n_br_sale.pot

Commit:
[UPD] Update l10n_br_stock_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_stock_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_stock_account/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_sale
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_sale/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_purchase
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_purchase/

OCA-git-bot committed:
Merge PR #3428 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3436 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal/

Tiago370 committed:
[IMP] l10n_br_cnpj_search: solves cnpj duplication problem in tests

Tiago370 committed:
[IMP] l10n_br_cnpj_search: transfer the ws mock to comon

Tiago370 committed:
[IMP] l10n_br_cnpj_search: transfer the serpro mock to common

OCA-git-bot committed:
Merge PR #3438 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[MIG] l10n_br_account_nfe: _render_qweb_* migration

CristianoMafraJunior committed:
[FIX] l10n_br_nfe: name template

OCA-git-bot committed:
Merge PR #3437 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3439 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

Tiago370 committed:
[IMP] l10n_br_ie_search: change vcr mocks for unittest mocks.

OCA-git-bot committed:
Merge PR #3440 into 16.0

Signed-off-by antoniospneto

Commit:
[UPD] Update l10n_br_account.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account/

OCA-git-bot committed:
Merge PR #3448 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

antoniospneto commented:
@rvalyi 

Apenas teria que ver a questão dos filtros do CNPJ/CPF o campo `cnpj_cpf` agora ele não é mais persistido no banco de dados, então ele não pode ser mais usado no dominio do filtro que atualmente está assim:

```
                <field
                    name="cnpj_cpf"
                    filter_domain="['|', ('cnpj_cpf', 'ilike', self), ('cnpj_cpf_stripped', 'ilike', self)]"
                />
```

Vejo duas soluções:

Solução 1: Marcar o campo `cnpj_cpf` como `stored=true`
Solução 2: Utilizar os campos `vat` e `l10n_br_cpf_code` no lugar de `cnpj_cpf` na estrutura do dominio do filtro.

Obs: esse mesmo filtro é aplicado em diversos modelos: contato, pedido, fatura, crm. é preciso aplicar a alteraçaõ em todos

antoniospneto reviewed (changes requested):
Concordo que a alteração é bem-vinda para manter a consistência com o Odoo Nativo.
Apenas precisamos verificar a questão dos filtros que comentei acima.

Tiago370 committed:
[FIX] equal Estadual Inscription detection

Tiago370 committed:
[IMP] add tests.

OCA-git-bot committed:
Merge PR #3450 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

rvalyi committed:
[REF] spec_driven_model: pre-commit fixes

OCA-git-bot committed:
Merge PR #3444 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_account_nfe.pot

Commit:
[UPD] Update l10n_br_nfe_spec.pot

Commit:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe_spec
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe_spec/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_account_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_account_nfe/

OCA-git-bot committed:
Merge PR #3459 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_nfe.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_nfe
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_nfe/

OCA-git-bot committed:
Merge PR #3457 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_fiscal_closing.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-l10n_br_fiscal_closing
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-l10n_br_fiscal_closing/

Tiago370 committed:
[IMP] l10n_br_cnpj_search: default_get from res.partner

OCA-git-bot committed:
Merge PR #3454 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update spec_driven_model.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-16.0/l10n-brazil-16.0-spec_driven_model
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-16-0/l10n-brazil-16-0-spec_driven_model/

OCA-git-bot committed:
Merge PR #3289 into 16.0

Signed-off-by rvalyi

Commit:
[UPD] Update l10n_br_crm_cnpj_search.pot

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[FIX] l10n_br_fiscal: tree view field

OCA-git-bot committed:
Merge PR #3468 into 16.0

Signed-off-by mileo

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[FIX] l10n_br_account: tree view field

CristianoMafraJunior committed:
[FIX] l10n_br_fiscal_edi: tree view field

CristianoMafraJunior committed:
[FIX] l10n_br_fiscal: tree view field

OCA-git-bot committed:
Merge PR #3470 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

crsilveira committed:
[FIX] l10n_br_fiscal: ncm starting with 09 conflicts with service type

OCA-git-bot committed:
Merge PR #3473 into 16.0

Signed-off-by antoniospneto

OCA-git-bot committed:
[BOT] post-merge updates

CristianoMafraJunior committed:
[FIX] l10n_br_base: form padding 0

OCA-git-bot committed:
Merge PR #3477 into 16.0

Signed-off-by rvalyi

OCA-git-bot committed:
[BOT] post-merge updates

rvalyi commented:
vale a pena ver que um refator importante do VAT/CNPJ que tinha sido feito pelo @marcelsavegnago na 14.0 ta atualmente faltando na 16.0 e por isso certamente esse meu PR ta incompleto:

https://github.com/OCA/l10n-brazil/issues/3487#issue-2655356105

cc @antoniospneto @renatonlima
